server/utils.go

```diff
@@ -4,9 +4,64 @@
 package main
 
 import (
+	"fmt"
+	"os"
+
 	"github.com/skip2/go-qrcode"
 )
 
+/*
+	Seals the given key with the TPM Storage Root Key
+	and stores it under two files named after the given
+	uuid. If those files already exists, an error is
+	returned.
+*/
+func storeKey(uuid string, key []byte) error {
+	var priv, pub []byte
+	var err error
+
+	if priv, pub, err = TPM2_Seal(key); err != nil {
+		return err
+	}
+	if err = os.MkdirAll(ULTRABLUE_KEYS_PATH, os.ModeDir); err != nil {
+		return err
+	}
+	if _, err = os.Stat(ULTRABLUE_KEYS_PATH + uuid); err == nil {
```

Comment:
Note that there is a small race condition (TOCTOU). The proper way to do this atomically is os.Open(O_CREAT|O_EXCL) which will return an error if the file exists and create it otherwise. Let's assume though that we have a single process writing to those files at a time and not bother with it.

Reply:
But maybe add a comment saying that we are aware of the race-condition and decide to ignore it?